internal/datasource/config/nginx_config_parser.go

```diff
@@ -784,6 +831,46 @@ func (ncp *NginxConfigParser) socketClient(socketPath string) *http.Client {
 	}
 }
 
+// prepareHTTPClient handles TLS config
+func (ncp *NginxConfigParser) prepareHTTPClient(ctx context.Context) (*http.Client, error) {
+	httpClient := http.DefaultClient
+	caCertLocation := ncp.agentConfig.DataPlaneConfig.Nginx.APITls.Ca
+
+	if caCertLocation != "" && ncp.agentConfig.IsDirectoryAllowed(caCertLocation) {
+		slog.DebugContext(ctx, "Reading CA certificate", "file_path", caCertLocation)
+		caCert, err := os.ReadFile(caCertLocation)
+		if err != nil {
+			return nil, err
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+
+		httpClient = &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					RootCAs:    caCertPool,
+					MinVersion: tls.VersionTLS13,
+				},
+			},
+		}
+	}
+
+	return httpClient, nil
+}
+
+// Populate the CA cert location based ondirectory allowance.
+func (ncp *NginxConfigParser) getCACertLocation(ctx context.Context) string {
```

Comment:
```suggestion
func (ncp *NginxConfigParser) caCertLocation(ctx context.Context) string {
```
Go naming conventions recommend not using `get` in naming